core/src/main/java/org/frankframework/stream/Message.java

```diff
@@ -851,7 +838,16 @@ public static String asString(Object object, String defaultCharset) throws IOExc
 		if (object instanceof String) {
 			return (String) object;
 		}
-		return Message.asMessage(object).asString(defaultCharset);
+		if (object instanceof Message) {
+			return ((Message) object).asString(defaultCharset);
+		}
+		if (object instanceof MessageWrapper) {
+			return ((MessageWrapper<?>) object).getMessage().asString(defaultCharset);
```

Comment:
I think that instead of the default charset you should look at the charset that the current message has? The charset passed to this message is the charset used to _decode_ the message and the message itself might already store another charset for the data it contains.

Reply:
Valid point. Didn't think about that and no unit tests fails on this topic..

Reply:
Discussed with Niels and refactored this a bit. We removed the extra static methods with `defautCharset` completely. It is not needed, since it is detected and managed elsewhere.